cocos/renderer/CCTexture2D.h

```diff
@@ -339,11 +339,19 @@ class CC_DLL Texture2D : public Ref
     void setAliasTexParameters();
 
 
-    /** Generates mipmap images for the texture.
+    /** DEPRECATED, use setAutoGenerateMipmap(before call init)
+    Generates mipmap images for the texture.
     It only works if the texture size is POT (power of 2).
     @since v0.99.0
     */
-    void generateMipmap();
+    CC_DEPRECATED_ATTRIBUTE void generateMipmap(){ generateMipmapInternal(); }
+
+    /** Need generate mipmap images for the texture.
+    Call before init
+    It only works if the texture size is POT (power of 2).
+    @since v3.11
+    */
+    void setAutoGenerateMipmap(bool generateMipmaps);
```

Comment:
You not understend.
`CC_DEPRECATED_ATTRIBUTE void generateMipmap(){ generateMipmapInternal(); }`
It works also as before.

If setAutoGenerateMipmap rename to generateMipmap, it break compatibility.